### PR TITLE
Feature/maven tutorial

### LINKED ISSUE #39: feature(Tutorial 2): Build a Java app with Maven
- For discussion to replicate Tutorial [Build a Java app with Maven](https://www.jenkins.io/doc/tutorials/build-a-java-app-with-maven/)
- Uses  docker in docker 
- Pinging mentors @berviantoleo @gounthar @jmMeessen

### PR DESCRIPTION
**Pull request by ash-sxn**
- resolves #39 
- pinging mentors @gounthar @berviantoleo @jmMeessen

### CODE REVIEW

File: experimental_docker_compose_files/03_maven_tutorial/dockerfiles/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,9 @@
+FROM jenkins/jenkins:2.401.1-lts
```
Comment (gounthar) [inline experimental_docker_compose_files/03_maven_tutorial/dockerfiles/Dockerfile:1]:
```suggestion
FROM jenkins/jenkins:2.401.2-lts
```

File: experimental_docker_compose_files/03_maven_tutorial/dockerfiles/agent/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    && tar xzf "/tmp/apache-maven-${MAVEN_VERSION}-bin.tar.gz" -C /opt/ \
+    && rm "/tmp/apache-maven-${MAVEN_VERSION}-bin.tar.gz" \
+    && ln -s /opt/apache-maven-${MAVEN_VERSION} /opt/maven \
+    && ln -s /opt/maven/bin/mvn /usr/bin/mvn \
+    && mkdir -p /etc/profile.d \
+    && echo "export JAVA_HOME=$JAVA_HOME \n \
+             export M2_HOME=/opt/maven \n \
+             export PATH=${M2_HOME}/bin:${PATH}" > /etc/profile.d/maven.sh
+ENV M2_HOME="/opt/maven"
+ENV PATH="${M2_HOME}/bin/:${PATH}"
```
Comment (gounthar) [inline experimental_docker_compose_files/03_maven_tutorial/dockerfiles/agent/Dockerfile:22]:
```suggestion
ENV M2_HOME="/opt/maven"
ENV PATH="${M2_HOME}/bin/:${PATH}"
RUN echo "PATH=${PATH}" >> /etc/environment
```

**Review by gounthar — CHANGES_REQUESTED** [review]
The Maven job is missing, right?
See https://github.com/ash-sxn/GSoC-2023-docker-based-quickstart/blob/368ddf627cb55950e020655b9e84c4d2008e0693/experimental_docker_compose_files/maven/dockerfiles/jobs/simple-java-maven-app/config.xml .

File: experimental_docker_compose_files/03_maven_tutorial/dockerfiles/plugins.txt [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+ws-cleanup:0.45
+ant:487.vd79d090d4ea_e
+ssh-slaves:2.877.v365f5eb_a_b_eec
+gradle:2.8
+workflow-aggregator:596.v8c21c963d92d
+github-branch-source:1728.v859147241f49
+pipeline-stage-view:2.32
+git:5.1.0
+matrix-auth:3.1.8
+mailer:457.v3f72cb_e015e5
```
Comment (gounthar) [inline experimental_docker_compose_files/03_maven_tutorial/dockerfiles/plugins.txt:16]:
```suggestion
mailer:457.v3f72cb_e015e5
pipeline-graph-view:191.vc6da_9d3eb_70a
```

File: experimental_docker_compose_files/03_maven_tutorial/fork-and-clone.sh [hunk 4]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+MAVEN_REPO="jenkins-docs/simple-java-maven-app"
+# Extract the repository name from MAVEN_REPO
+REPO_NAME=${MAVEN_REPO##*/}
+
+# Create a directory on the host machine to store the cloned repository
+CLONE_DIR=$(pwd)/$REPO_NAME
```
Comment (gounthar) [inline experimental_docker_compose_files/03_maven_tutorial/fork-and-clone.sh:8]:
```suggestion
CLONE_DIR=$(pwd)/../../$REPO_NAME
```

**Comment by ash-sxn** [comment]
@gounthar
Yes, I've not added the job. I was thinking setting that up will be a good part for the beginner users.
I will add the `fork-and-clone.sh` to the init script, I also think It will be better to do that with init script 
and I was also wondering about the slides for the presentation. I'll share them on Gitter after creating them. for feedback.

**Review by gounthar — CHANGES_REQUESTED** [review]
Please see all the files that end up with a 🚫 or equivalent.
That means an empty end line is missing. That is not [POSIX compliant](https://www.techtarget.com/whatis/definition/POSIX-Portable-Operating-System-Interface).
Could you please correct it?

File: jenkins_init.sh [hunk 5]
```diff
@@ -93,7 +93,7 @@ check_docker_compose() {
 generate_ssh_keys() {
   local tutorial_path=$1
   echo "generating new ssh keys"
-  sh $tutorial_path/keygen.sh $tutorial_path
+  bash $tutorial_path/keygen.sh $tutorial_path
```
Comment (gounthar) [inline jenkins_init.sh:96]: Why not call the script directly?
Are there any challenges or considerations that I may not be aware of?

**Review by gounthar — APPROVED** [review]

**Pull request merged**